commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPAttachmentFileEntryLocalServiceImpl.java

```diff
@@ -587,6 +634,31 @@ protected void checkCPAttachmentFileEntriesByExpirationDate()
 		}
 	}
 
+	protected long getGroupId(long classNameId, long classPK)
+		throws PortalException {
+
+		long cpDefinitionClassNameId = classNameLocalService.getClassNameId(
+			CPDefinition.class);
+		long assetCategoryClassNameId = classNameLocalService.getClassNameId(
+			AssetCategory.class);
+
+		if (classNameId == assetCategoryClassNameId) {
```

Comment:
@alecsloan this is going to limit this service working just with categories and products, let's talk later for a better solution, we do need to externalize those media staff, as we do have requirements for attachments to orders and shipment as well

Reply:
breaking change, we do need to pass the groupId in the method. @fafonso be aware that this will impact your customization

Reply:
@marco-leo Please be aware I've pushed the requested change. Compilation is successful but I didn't have time today to test out the changes.